ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -149,6 +171,7 @@ export default class ConfirmPageContainer extends Component {
             />
           )}
         </ConfirmPageContainerHeader>
+        <div>{this.maybeRenderAddContact()}</div>
```

Comment:
my preference would be to determine whether to show the dialog and assign to variable. then render inline based on that variable. Pseudocode below.

```js
render() {
  const showAddToAddressDialog = contact.name === undefined && toAddress !== undefined;
  return (
   <>{showAddToAddressDialog && (
      <Dialog
        type="message"
        className="send__dialog"
        onClick={showAddToAddressBookModal}
      >
        {t('newAccountDetectedDialogMessage')}
      </Dialog>
   )}
   </>
  );
```